### PR TITLE
docs: Custom icon usage instructions for iOS share preview added to example

### PR DESCRIPTION
**Pull request by sercanuste**
Example for: #676

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -223,10 +223,11 @@ import Share from 'react-native-share';
 const url = 'https://awesome.contents.com/';
 const title = 'Awesome Contents';
 const message = 'Please check this out.';
+const icon = 'data:<data_type>/<file_extension>;base64,<base64_data>';
```
Comment (MateusAndrade) [inline README.md:226]: Only works with base64?

**Comment by MateusAndrade** [comment]
Hey @sercanuste , would you mind showing an example of that on iOS? I haven't tested the new share system on iOS very often. This would be nice to have an image example, even if juts on this PR. 🤗

File: README.md (hunk 1 above)
Comment (sercanuste) [reply README.md:226]: I tested with url. It overrides link metadata when image from url loaded.

**Comment by sercanuste** [comment]
Hi,

First image is the default behavior with message 
```jsx
{
  message: `${message} ${url}`
}
```
image

And this is with activityItemSources
```jsx
{
    activityItemSources: [
        {
            placeholderItem: {
                type: 'url',
                content: icon
            },
            item: {
                default: {
                    type: 'text',
                    content: `${message} ${url}`
                },
            },
            linkMetadata: {
                title: message,
                icon: icon
            }
        }
    ],
}
```
image

**Comment by sercanuste** [comment]
I found out while testing that using activityItemSources like this removes default text icon and shows app icon on share preview. This can be real answer for #676 and PR is for custom icon alternative.

```jsx
activityItemSources: [
    {
        placeholderItem: {
            type: 'text',
            content: `${message} ${url}`
        },
        item: {
            default: {
                type: 'text',
                content: `${message} ${url}`
            },
        },
        linkMetadata: {
            title: message,
        }
    }
],
```

File: README.md (hunk 1 above)
Comment (MateusAndrade) [reply README.md:226]: Maybe we can add something like:

```suggestion
const icon = 'data:<data_type>/<file_extension>;base64,<base64_data>' or "remote-image-url";
```

This should work?

Comment (sercanuste) [reply README.md:226]: When I use remote url it loads title and url from internet and overrides linkMetadata.
```jsx
const icon = 'https://www.sercanuste.com/img/logo/BA.png';
```
IMG_162F379A8183-1

**Pull request merged**